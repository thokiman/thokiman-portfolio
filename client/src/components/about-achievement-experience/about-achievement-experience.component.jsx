import React from "react";

import {
  AchievementExperienceProjectContainer,
  AchievementExperienceProjectTitle,
  AchievementExperienceProject,
  DateAchievementExperienceProject,
  NameAchievementExperienceProject,
  ImageAchievementExperience,
  ImageAchievementExperienceContainer,
  DescriptionAchievementExperienceProject,
} from "./about-achievement-experience.styles";

const AboutAchievementProject = ({ lastExperience }) => {
  return (
    <AchievementExperienceProjectContainer>
      <AchievementExperienceProjectTitle>
        Experience of Achievement
      </AchievementExperienceProjectTitle>
      {lastExperience.map(({ id, name, description, date, imagePath }) => {
        return (
          <AchievementExperienceProject key={id}>
            <NameAchievementExperienceProject>
              {name}
            </NameAchievementExperienceProject>
            <DateAchievementExperienceProject>
              {date}
            </DateAchievementExperienceProject>
            <ImageAchievementExperienceContainer key={id}>
              {imagePath.map(({ id, field, iconPath }) => {
                return (
                  <ImageAchievementExperience
                    key={id}
                    src={iconPath}
                    alt={field}
                  />
                );
              })}
            </ImageAchievementExperienceContainer>
            <DescriptionAchievementExperienceProject>
              {description}
            </DescriptionAchievementExperienceProject>
          </AchievementExperienceProject>
        );
      })}
    </AchievementExperienceProjectContainer>
  );
};

export default AboutAchievementProject;
