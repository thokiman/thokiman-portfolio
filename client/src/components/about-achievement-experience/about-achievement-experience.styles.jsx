import styled, { css } from "styled-components";

export const AchievementExperienceProjectContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 95%;
  margin: 0 auto;
`;

AchievementExperienceProjectContainer.displayName =
  "AchievementExperienceProjectContainer";

export const AchievementExperienceProjectTitle = styled.div`
  font-size: 350%;
  padding: 10px;
  color: #e3e1e4;
  letter-spacing: 0.04em;
  text-align: left;
  display: flex;
  align-items: center;
  width: 100%;
  @media only screen and (max-width: 800px) {
    font-size: 14vw;
  }
`;

AchievementExperienceProjectTitle.displayName =
  "AchievementExperienceProjectTitle";

export const AchievementExperienceProject = styled.div`
  width: 100%;
  height: 100%;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-auto-rows: repeat(3, 1fr);
  gap: 10px 10px;
  grid-template-areas:
    "name-achievement-experience-project name-achievement-experience-project date-achievement-experience-project date-achievement-experience-project"
    "image-achievement-experience-container image-achievement-experience-container description-achievement-experience-project description-achievement-experience-project"
    "image-achievement-experience-container image-achievement-experience-container description-achievement-experience-project description-achievement-experience-project";
  margin-bottom: 50px;
  border-bottom: #74c6d4 solid 1px;
`;

AchievementExperienceProject.displayName = "AchievementExperienceProject";

const AchievementExpericenceTextStyles = css`
  text-decoration: none;
  color: #e3e1e4;
  letter-spacing: 0.09em;
  display: flex;
  justify-content: flex-start;
  align-items: center;
`;

export const NameAchievementExperienceProject = styled.div`
  grid-area: name-achievement-experience-project;
  width: 100%;
  font-size: 250%;
  ${AchievementExpericenceTextStyles}
  margin-bottom:10px;
  @media only screen and (max-width: 800px) {
    font-size: 7vw;
  }
`;

NameAchievementExperienceProject.displayName =
  "NameAchievementExperienceProject";

export const DateAchievementExperienceProject = styled.div`
  grid-area: date-achievement-experience-project;
  width: 100%;
  border-left: #74c6d4 solid 1px;
  font-size: 190%;
  padding: 0 20px;
  @media only screen and (max-width: 800px) {
    font-size: 6vw;
  }
  ${AchievementExpericenceTextStyles}
`;

DateAchievementExperienceProject.displayName =
  "DateAchievementExperienceProject";

export const ImageAchievementExperienceContainer = styled.div`
  grid-area: image-achievement-experience-container;
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  padding-right: 5px;
`;

ImageAchievementExperienceContainer.displayName =
  "ImageAchievementExperienceContainer";

export const ImageAchievementExperience = styled.img`
  width: 100%;

  background-size: contain;
  opacity: 0.85;
`;

ImageAchievementExperience.displayName = "ImageAchievementExperience";

export const DescriptionAchievementExperienceProject = styled.div`
  grid-area: description-achievement-experience-project;
  width: 100%;
  font-size: 120%;
  line-height: 1.4em;
  text-align: left;
  ${AchievementExpericenceTextStyles}\
  margin-bottom: 10px;
  @media only screen and (max-width: 800px) {
    font-size: 5vw;
  }
`;

DescriptionAchievementExperienceProject.displayName =
  "DescriptionAchievementExperienceProject ";
