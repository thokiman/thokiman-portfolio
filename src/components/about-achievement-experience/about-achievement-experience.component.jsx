import React from "react";

import "./about-achievement-experience.styles.scss";

const AboutAchievementProject = ({ lastExperience }) => {
  return (
    <div className="achievement-experience-project-container">
      <div className="achievement-experience-project-title">
        Experience of Achievement
      </div>
      {lastExperience.map(({ id, name, description, date, imagePath }) => {
        return (
          <div key={id} className="achievement-experience-project">
            <div className="name-achievement-experience-project">{name}</div>
            <div className="date-achievement-experience-project">{date}</div>
            <div key={id} className="image-achievement-experience-container">
              {imagePath.map(({ id, field, iconPath }) => {
                return (
                  <img
                    key={id}
                    src={iconPath}
                    alt={field}
                    className="image-achievement-experience"
                  />
                );
              })}
            </div>
            <div className="description-achievement-experience-project">
              {description}
            </div>
          </div>
        );
      })}
    </div>
  );
};

export default AboutAchievementProject;
