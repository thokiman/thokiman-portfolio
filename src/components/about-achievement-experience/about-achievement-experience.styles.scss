.achievement-experience-project-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 95%;
    margin: 0 auto;

    .achievement-experience-project-title {
        font-size: 350%;
        padding: 10px;
        color: #e3e1e4;
        letter-spacing: 0.04em;
        text-align: left;
        display: flex;
        align-items: center;
        width: 100%;
    }

    .achievement-experience-project {

        width: 85%;
        height: 100%;
        display: grid;
        grid-template-columns: repeat(4, 1fr);
        grid-auto-rows: repeat(3, 1fr);
        gap: 10px 10px;
        grid-template-areas:
            "name-achievement-experience-project name-achievement-experience-project date-achievement-experience-project date-achievement-experience-project"
            "image-achievement-experience-container image-achievement-experience-container description-achievement-experience-project description-achievement-experience-project"
            "image-achievement-experience-container image-achievement-experience-container description-achievement-experience-project description-achievement-experience-project";

        margin-bottom: 50px;

        border-bottom: #74c6d4 solid 1px;

        .name-achievement-experience-project {

            width: 100%;
            text-decoration: none;
            grid-area: name-achievement-experience-project;
            font-size: 350%;
            ;
            color: #e3e1e4;
            letter-spacing: 0.09em;

            display: flex;
            align-items: center;
            justify-content: flex-start;


        }

        .date-achievement-experience-project {
            border-left: #74c6d4 solid 1px;
            width: 85%;
            text-decoration: none;
            grid-area: date-achievement-experience-project;
            font-size: 190%;
            padding: 0 20px;
            color: #e3e1e4;
            letter-spacing: 0.09em;

            display: flex;
            align-items: center;
            justify-content: flex-start;


        }

        .image-achievement-experience-container {
            grid-area: image-achievement-experience-container;

            width: 94%;
            height: 89%;
            display: flex;
            align-items: center;
            justify-content: center;
            padding-right: 5px;



            .image-achievement-experience {
                width: 90%;
                height: 100%;
                background-size: cover;
                opacity: 0.85;
            }
        }

        .description-achievement-experience-project {


            width: 300px;
            grid-area: description-achievement-experience-project;
            font-size: 120%;
            display: flex;
            align-items: center;
            text-align: left;
            line-height: 1.2em;
            color: #e3e1e4;
            letter-spacing: 0.09em;




        }
    }
}