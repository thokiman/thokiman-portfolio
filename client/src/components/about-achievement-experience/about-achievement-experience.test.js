import React from "react";
import { shallow } from "enzyme";

import { AboutAchievementProject } from "./about-achievement-experience.component";

describe("AboutAchievementProject", () => {
  let wrapper;
  const mockArray = [];
  beforeEach(() => {
    const mockProps = {
      lastExperience: mockArray,
    };
    wrapper = shallow(<AboutAchievementProject {...mockProps} />);
  });
  it("should render AboutAchievementProject component", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("shoud render an equal props of AboutAchievementProject as lastExperience prop", () => {
    expect(wrapper.exists("AchievementExperienceProject")).toBe(false);
  });
});
