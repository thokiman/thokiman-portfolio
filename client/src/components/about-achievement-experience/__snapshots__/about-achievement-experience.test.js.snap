// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutAchievementProject should render AboutAchievementProject component 1`] = `ShallowWrapper {}`;
